src/differentiation.jl

```diff
@@ -18,49 +18,34 @@ function jacobian(
         ts::V;
         ε = 1e-6
 ) where {G <: Meshes.Geometry, V <: Union{AbstractVector, Tuple}}
-    T = eltype(ts)
+    Dim = Meshes.paramdim(geometry)
+    if Dim != length(ts)
+        throw(ArgumentError("ts must have same number of dimensions as geometry."))
+    end
 
-    # Get the partial derivative along the n'th axis via finite difference approximation
-    #   where ts is the current parametric position (εv is a reusable buffer)
-    function ∂ₙr!(εv, ts, n)
+    T = eltype(ts)
+    ε = T(ε)
+
+    # Get the partial derivative along the n'th axis via finite difference
+    #   approximation, where ts is the current parametric position
+    function ∂ₙr(ts, n)
+        # Build left/right parametric coordinates with non-allocating iterators 
+        left = Iterators.map(it -> it[1] == n ? it[2] - ε : it[2], enumerate(ts))
```

Comment:
```suggestion
        left = Iterators.map(it -> ((i, t) = it; i == n ? t - ε : t), enumerate(ts))
```
I saw your question. I'm not sure if there is an optimal solution, but this is another option. I don't see a solution, where you can pass something to the anonymous function, which is not a `Tuple`.

Reply:
I finally figured it out and just commit updates to these lines. Looks like the parser interprets the extra parenthesis in `((i, t)) -> ...` as superfluous, and needs to see a trailing comma `((i, t),) -> ...` to understand that this is a tuple being destructured.

Reply:
Ah, that's nice.